### PR TITLE
[core] Remove gcs addr updater in core worker.

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?
Since we are using domain name resolution to get the new address of GCS, gcs addr updator is not necessary any more. This PR removed that.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by mwtian — APPROVED** [review]

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
Btw cc @WangTaoTheTonic for the visibility

**Comment by iycheng** [comment]
Interesting..., it seems there is another bug depending on this CHECK failure to crash worker.

**Comment by iycheng** [comment]
@rkooo567  @mwtian  could you take another look. The issue here is that after removing this, worker won't fail due to fail to get gcs address. Raylet sometimes is a zombie and we need to check that as well.

**Comment by rkooo567** [comment]
Do you happen to know why the raylet is zomebie in this case?

**Comment by iycheng** [comment]
> Do you happen to know why the raylet is zomebie in this case? 

The reason is because the process which starts it doesn't claim for the result. It should be the pytest. Another way might be just to fix this test. How do you think about that?

**Comment by rkooo567** [comment]
I see. So to make sure I fully understood;

Before:
Pytest starts a raylet -> GCS crashed -> Raylet crashed -> worker was originally killed by gcs address updater.

Now:
Since there's no GCS address updater, we should rely on raylet being killed (the worker should kill itself). However, raylet is a zombie process.

Is this correct?

File: src/ray/util/process.cc [hunk 1]
```diff
@@ -612,11 +613,33 @@ bool IsProcessAlive(pid_t pid) {
...
   }
   return false;
-#else
+#elif defined __APPLE__
   if (kill(pid, 0) == -1 && errno == ESRCH) {
     return false;
   }
   return true;
+#else
+  if (kill(pid, 0) == -1 && errno == ESRCH) {
```
Comment (rkooo567) [inline src/ray/util/process.cc:622]: Can we have cpp unit test? I guess it should be pretty easy to verify it if it works on both Mac & Linux correctly?

Comment (iycheng) [reply src/ray/util/process.cc:622]: No, only for Linux here. Mac doesn't have proc and Mac test not fail as well.

**Review by mwtian — APPROVED** [review]

**Comment by iycheng** [comment]
@rkooo567  this is what I mean. It's a bug from my understanding. Previously, the worker needs to wait for x seconds to fail.

**Comment by iycheng** [comment]
Let me add a simple unit test for this.

File: src/ray/util/process.cc [hunk 2]
```diff
@@ -612,11 +613,33 @@ bool IsProcessAlive(pid_t pid) {
...
+  snprintf(pbuf, sizeof(pbuf), "/proc/%d/stat", (int)pid);
+  FILE *fpstat = fopen(pbuf, "r");
+  if (!fpstat) {
+    return false;
+  }
+
+  int rpid = 0;
+  char rcmd[32];
+  char rstatc = 0;
+  RAY_CHECK(fscanf(fpstat, "%d %30s %c", &rpid, rcmd, &rstatc) == 3);
```
Comment (scv119) [inline src/ray/util/process.cc:639]: nit: should we tolerate the read failure case?

**Review by scv119 — APPROVED** [review]

File: src/ray/util/process.cc (hunk 2 above)
Comment (iycheng) [reply src/ray/util/process.cc:639]: This is something defined in linux kernel: https://linux.die.net/man/5/proc which is very stable.

In this case, I think if we fail, we should fail early.

**Comment by iycheng** [comment]
Mac has the same issue. But I don't know how to detect that in mac.

**Comment by rkooo567** [comment]
@iycheng hmm should we just send a RPC instead of checking process alive?

**Comment by rkooo567** [comment]
but I am okay with just fixing tests + adding zombie process detecting logic to this PR (with an open issue)

**Comment by iycheng** [comment]
@rkooo567  My concern is that rpc seems more complicated.
But anyway we have worker talking with local rayelt, maybe it's ok to add it there.

**Comment by iycheng** [comment]
mac started to fail now. let me find another way to fix this. 😅

**Comment by iycheng** [comment]
Pretty bad... more tests failed. Investigating.

**Comment by rkooo567** [comment]
Lmk if you'd like to do pair debugging or sth!

**Comment by iycheng** [comment]
@rkooo567  Thanks for the offer. But it's already fixed (wait if we remote node in cluster utils). I'm just waiting for the tests.

**Pull request merged**